central/sensor/service/pipeline/nodeinventory/two_pipelines_test.go

```diff
@@ -0,0 +1,404 @@
+					m.riskStorage.EXPECT().UpsertRisk(gomock.Any(), gomock.Any()).Times(2).Return(nil),
+					m.nodeDatastore.EXPECT().UpsertNode(gomock.Any(), gomock.Any()).Return(nil),
+
+					// check what got stored in the DB
+					m.nodeDatastore.EXPECT().GetNode(gomock.Any(), gomock.Eq(nodeID)).AnyTimes().Return(nodeWithScanWithKernelV1, true, nil),
+				)
+			},
+			wantNodeExists:            true,
+			wantKernelVersionNode:     "v1",
+			wantKernelVersionNodeScan: "v1",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			tt.mocks = usedMocks{
+				clusterStore:      clusterDatastoreMocks.NewMockDataStore(ctrl),
+				nodeDatastore:     nodeDatastoreMocks.NewMockDataStore(ctrl),
+				cveDatastore:      nodeCVEDataStoreMocks.NewMockDataStore(ctrl),
+				deploymentStorage: mocks.NewMockDataStore(ctrl),
+				imageStorage:      imageStoreMock.NewMockDataStore(ctrl),
+				riskStorage:       riskStoreMock.NewMockDataStore(ctrl),
+				updater:           updaterMocks.NewMockUpdater(ctrl),
+			}
+			tt.riskManager = manager.New(
+				tt.mocks.nodeDatastore,
+				tt.mocks.deploymentStorage,
+				tt.mocks.imageStorage,
+				tt.mocks.riskStorage,
+				&mockNodeScorer{},
+				&mockComponentScorer{},
+				&mockDeploymentScorer{},
+				&mockImageScorer{},
+				&mockComponentScorer{},
+
+				ranking.ClusterRanker(),
+				ranking.NamespaceRanker(),
+				ranking.ComponentRanker(),
+				ranking.NodeComponentRanker(),
+
+				tt.mocks.updater,
+			)
+			creator := func() (string, scanners.NodeScannerCreator) {
+				return "fake", func(integration *storage.NodeIntegration) (types.NodeScanner, error) {
+					return &fakeNodeScanner{}, nil
+				}
+			}
+			tt.enricher = nodeEnricher.NewWithCreator(tt.mocks.cveDatastore, metrics.CentralSubsystem, creator)
+			err := tt.enricher.UpsertNodeIntegration(&storage.NodeIntegration{
+				Id:   "1",
+				Name: "dummy-scanner",
+				Type: "fake",
+				IntegrationConfig: &storage.NodeIntegration_Clairify{Clairify: &storage.ClairifyConfig{
+					Endpoint:           "abc",
+					GrpcEndpoint:       "",
+					NumConcurrentScans: 0,
+				}},
+			})
+			assert.NoError(t, err)
+
+			if tt.setUpMocks != nil {
+				tt.setUpMocks(t, &tt.mocks)
+			}
+			pNode := nodes.NewPipeline(tt.mocks.clusterStore, tt.mocks.nodeDatastore, tt.enricher, tt.riskManager)
+			pNodeInv := newPipeline(tt.mocks.clusterStore, tt.mocks.nodeDatastore, tt.enricher, tt.riskManager)
+
+			var lastErr error
+			for i, op := range tt.operations {
+				t.Logf("Running operation %d of %d", i+1, len(tt.operations))
+				lastErr = op(t, pNode, pNodeInv)
+			}
+			if tt.wantErr != "" {
+				assert.ErrorContainsf(t, lastErr, tt.wantErr, "Run() error = %v, wantErr = %q", lastErr, tt.wantErr)
+			}
+
+			node, found, err := tt.mocks.nodeDatastore.GetNode(context.Background(), nodeID)
+			assert.Equal(t, tt.wantNodeExists, found)
+			assert.NoError(t, err)
```

Comment:
Suggestion: you can use `require.*` functions instead of assert if you want the test to stop on failure. 

```go
require.NoError(t, err)
// test will not get here if either assertion failed
```